### PR TITLE
feat: add query command

### LINKED ISSUE #489: Query User Information (Wallet Address, Multiplier)
I have been doing some manual payouts lately and realized that sometimes it's nice to be able to quickly identify a user's `multiplier` and `wallet` address without superuser access to the global bot database!

We should have a new command to view a user's settings. 

`/query @user`

Unless there is a better, more expressive command name than "query" 

---

> I'll need to send some xDAI to both participants because this was my fault for not ensuring that the bounty is valid. 

> https://gnosisscan.io/tx/0x3162608e9e18f71b4b29b0780904b664d160d0d79b0f761569f0d590b5f16044 @me505 
> https://gnosisscan.io/tx/0xbb9e74ab8951f5cc45fb38dd1d56f7e04e3c0ba6db98bfbaf58bfe63043d950a @AnakinSkywalkeer

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquity-dollar/issues/562#issuecomment-1630854865_

### PR DESCRIPTION
**Pull request by me505**
Resolves #489

### CODE REVIEW

**Comment by me505** [comment]
@0xcodercrane what else is needed to make a command other than what i did?

File: src/handlers/comment/handlers/query.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { getBotContext, getLogger } from "../../../bindings";
+import { Payload } from "../../../types";
+import { getWalletAddress, getWalletMultiplier } from "../../../adapters/supabase";
+
+export const multiplier = async (body: string) => {
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/query.ts:5]: `multiplier` doesn't make sense for query functionality.

File: src/handlers/comment/handlers/query.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    logger.info(`Skipping '/query' because of no issue instance`);
+    return;
+  }
+
+  const regex = /\S+\s+@(\S+)/;
+  const matches = body.match(regex);
+  const user = matches?.shift()
+  if (user) {
+    const walletAddress = getWalletAddress(user)
+    const multiplier = getWalletMultiplier(user)
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/query.ts:24]: Both wallet address and multiplier are in the `wallets` table so should be able to get both with one request

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]

File: src/handlers/comment/handlers/query.ts (hunk 1 above)
Comment (me505) [reply src/handlers/comment/handlers/query.ts:5]: i forgot to change that 💀

File: src/handlers/comment/handlers/query.ts (hunk 2 above)
Comment (me505) [reply src/handlers/comment/handlers/query.ts]: done

File: src/handlers/comment/handlers/query.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  if (!issue) {
+    logger.info(`Skipping '/query' because of no issue instance`);
+    return;
+  }
+
+  const regex = /\S+\s+@(\S+)/;
+  const matches = body.match(regex);
+  const user = matches?.shift()
+  if (user) {
+    const { data: multiplierData, error: multiplierError } = await supabase.from('wallets').select('multiplier, address').eq("user_name", user).single();
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/query.ts:24]: we should create a helper function in adapters/supabase and use it here

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]

**Review by 0xcodercrane — APPROVED** [review]

**Comment by devpanther** [comment]
@0xcodercrane this is an unfinished command, it would never had passed the QA

The command wasn't added to this object https://github.com/ubiquity/ubiquibot/blob/6ad824269278ff8b17eb81ced96f1855a31dc125/src/handlers/comment/handlers/index.ts#L100, so basically it cannot trigger

**Comment by devpanther** [comment]
I think a hotfix should be made, the regex used is wrong as well

CP: @me505 @pavlovcik

**Comment by me505** [comment]
sorry about that, ill be waiting for the hotfix

**Comment by me505** [comment]
i have fixed the issues and now i am waiting for the hotfix, Thanks @devpanther for the catch

**Comment by devpanther** [comment]
> i have fixed the issues and now i am waiting for the hotfix, Thanks @devpanther for the catch

You have to make the hotfix PR

**Comment by me505** [comment]
oh i thought they open hotfix issues as i have seen one in this pr
sorry bout that

**Comment by pavlovcik** [comment]
@devpanther thanks for highlighting this. If helping with QA is at all of interest to you, we can chat about a collaboration to support @Draeieg

**Comment by devpanther** [comment]
> @devpanther thanks for highlighting this. If helping with QA is at all of interest to you, we can chat about a collaboration to support @Draeieg 

Sure @pavlovcik, it interests me.

**Comment by pavlovcik** [comment]
@Draeieg you should figure out how to delegate by communicating with @devpanther so that QA is being addressed in time.

**Pull request merged**